pmd-core/src/main/java/net/sourceforge/pmd/util/database/DBURI.java

```diff
@@ -420,95 +420,54 @@ public void setURL(String jdbcURL) {
      * @throws IOException
      */
     private void setFields() throws URISyntaxException, IOException {
-        if (url.startsWith("jdbc:")) {
-            // java.net.URI is intended for "normal" URLs
-            URI jdbcURI = new URI(getURL().substring(5));
-
-            LOG.debug("setFields - substr(jdbcURL,5):{}", getURL().substring(5));
-            dump("substr(jdbcURL,5)", jdbcURI);
-
-            // jdbc:subprotocol:subname
-            String[] uriParts = url.split(":");
-            for (String part : uriParts) {
-                LOG.trace("JDBCpart={}", part);
-            }
-
-            /*
-             * Expect jdbc : subprotocol [ : subname ] : connection details
-             * uriParts.length < 3 Error uriParts.length = 3 Driver information
-             * may be inferred from part[1] - the subprotocol uriParts.length >=
-             * 4 Driver information may be inferred from part[2]- the first part
-             * of the subname
-             */
-            if (3 == uriParts.length) {
-                subprotocol = uriParts[1];
-            } else if (4 <= uriParts.length) {
-                subprotocol = uriParts[1];
-                subnamePrefix = uriParts[2];
-            } else {
-                throw new URISyntaxException(getURL(), "Could not understand JDBC URL", 1);
-            }
-
-            LOG.debug("subprotocol={}'' subnamePrefix={}", subprotocol, subnamePrefix);
-
-            // Set values from DBType defaults
-            this.dbType = new DBType(subprotocol, subnamePrefix);
-
-            LOG.debug("DBType properties found at {} with {} properties.",
-                    dbType.getPropertiesSource(), dbType.getProperties().size());
-
-            LOG.trace("DBType properties are:- {}", dbType.getProperties());
-
-            if (null != dbType.getDriverClass()) {
-                this.driverClass = dbType.getDriverClass();
-            }
-
-            if (null != dbType.getCharacterSet()) {
-                this.characterSet = dbType.getCharacterSet();
-            }
-
-            if (null != dbType.getLanguages()) {
-                this.languages = dbType.getLanguages();
-            }
+        if (!url.startsWith("jdbc:")) {
+            throw new URISyntaxException(getURL(), "Could not understand JDBC URL", 1);
+        }
 
-            if (null != dbType.getSourceCodeTypes()) {
-                sourceCodeTypes = dbType.getSourceCodeTypes();
-            }
+        URI jdbcURI = new URI(getURL().substring(5));
+        LOG.debug("setFields - substr(jdbcURL,5):{}", getURL().substring(5));
+        dump("substr(jdbcURL,5)", jdbcURI);
 
-            LOG.debug("DBType other properties follow  ...");
+        String[] uriParts = url.split(":");
+        if (uriParts.length < 3 || uriParts.length >= 4) {
```

Comment:
Can be replaced with `uriParts.length != 3`. And if the condition is written this way, then `jdbc:oracle:oci:system/oracle@//192.168.100.21:1521/ORCL` string as the input for the DBURI constructor will not execute properly.

Please check the DBURITest for reference.